src/useq/_grid.py

```diff
@@ -350,4 +364,194 @@ def _offset_y(self, dy: float) -> float:
+    """
+
+    num_points: int
+    max_width: float
+    max_height: float
+    shape: Shape
+    random_seed: Optional[int] = None
+    allow_overlap: bool = True
+
+    @property
+    def is_relative(self) -> bool:
+        return True
+
+    def __iter__(self) -> Iterator[GridPosition]:  # type: ignore
+        func = _POINTS_GENERATORS[self.shape]
+
+        min_distance = (self.fov_width, self.fov_height)
+
+        for x, y in func(
+            self.num_points,
+            self.max_width,
+            self.max_height,
+            min_distance,
+            self.allow_overlap,
+            self.random_seed,
+        ):
+            yield GridPosition(x, y, 0, 0, True)
+
+    def num_positions(self) -> int:
+        return self.num_points
+
+
+def _random_points_in_ellipse(
+    num_points: int,
+    max_width: float,
+    max_height: float,
+    min_distance: Tuple[Optional[float], Optional[float]],
+    allow_overlap: bool,
+    random_seed: Optional[int],
+) -> Iterable[Tuple[float, float]]:
+    """Generate a random point around a circle with center (0, 0).
+
+    The point is within +/- radius_x and +/- radius_y at a random angle.
+    """
+    _iter = _get_iterator(Shape.ELLIPSE, num_points, random_seed)
+    points: list[Tuple[float, float]] = []
+
+    try:
+        while len(points) < num_points:
+            x0, y0, angle = next(_iter)
+            x, y = (
+                math.sqrt(x0) * (max_width / 2) * math.cos(angle * 2 * math.pi),
+                math.sqrt(y0) * (max_height / 2) * math.sin(angle * 2 * math.pi),
+            )
+            if _check_validity(allow_overlap, min_distance, points, x, y):
+                points.append((x, y))
+
+    except StopIteration:
+        _raise_warning(points)
+
+    return points
+
+
+def _random_points_in_rectangle(
+    num_points: int,
+    max_width: float,
+    max_height: float,
+    min_distance: Tuple[Optional[float], Optional[float]],
+    allow_overlap: bool,
+    random_seed: Optional[int],
+) -> Iterable[Tuple[float, float]]:
+    """Generate a random point around a rectangle with center (0, 0).
+
+    The point is within the bounding box (-width/2, -height/2, width, height)
+    """
+    _iter = _get_iterator(Shape.RECTANGLE, num_points, random_seed)
+    points: list[Tuple[float, float]] = []
+
+    try:
+        while len(points) < num_points:
+            x0, y0 = next(_iter)
+            x, y = (
+                (x0 * max_width) - (max_width / 2),
+                (y0 * max_height) - (max_height / 2),
+            )
+            if _check_validity(allow_overlap, min_distance, points, x, y):
+                points.append((x, y))
+
+    except StopIteration:
+        _raise_warning(points)
+
+    return points
+
+
+def _get_iterator(
+    shape: Shape, num_points: int, random_seed: Optional[int]
+) -> Iterator[Tuple[float, ...]]:
+    """Return an iterator of random numbers between 0 and 1 with size depending on the
+    `shape`.
+    """  # noqa: D205
```

Comment:
the checks are there for a reason :)  don't use noqa unless it's a very good reason.  these docstrings are not [pep 257 compliant](https://peps.python.org/pep-0257/)